test/Performance/BuildPerformanceTest.cs

```diff
@@ -0,0 +1,249 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Microsoft.Xunit.Performance;
+using Microsoft.DotNet.TestFramework;
+
+namespace Microsoft.DotNet.Tools.Builder.Tests
+{
+    public class BuildPerformanceTest : PerformanceTestBase
+    {
+        private static string SingleTargetApp = "SingleTargetApp";
+        private static string TwoTargetApp = "TwoTargetApp";
+
+        private static string[] SingleTargetGraph = new[]
+        {
+            "SingleTargetGraph/SingleTargetP0",
+            "SingleTargetGraph/SingleTargetP1",
+            "SingleTargetGraph/SingleTargetP2"
+        };
+
+        private static string[] TwoTargetGraph = new[]
+        {
+            "TwoTargetGraph/TwoTargetP0",
+            "TwoTargetGraph/TwoTargetP1",
+            "TwoTargetGraph/TwoTargetP2"
+        };
+
+        private static string[] TwoTargetGraphLarge = new[]
+        {
+            "TwoTargetGraphLarge/TwoTargetLargeP0",
+            "TwoTargetGraphLarge/TwoTargetLargeP1",
+            "TwoTargetGraphLarge/TwoTargetLargeP2",
+            "TwoTargetGraphLarge/TwoTargetLargeP3",
+            "TwoTargetGraphLarge/TwoTargetLargeP4",
+            "TwoTargetGraphLarge/TwoTargetLargeP5",
+            "TwoTargetGraphLarge/TwoTargetLargeP6"
+        };
+
+        [Benchmark]
+        public void BuildSingleProject_SingleTargetApp() => BuildSingleProject(CreateTestInstance(SingleTargetApp));
+        [Benchmark]
+        public void BuildSingleProject_TwoTargetApp() => BuildSingleProject(CreateTestInstance(TwoTargetApp));
+
+        public void BuildSingleProject(TestInstance instance)
+        {
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    Build(instance.TestRoot);
+                }
+                RemoveBin(instance.TestRoot);
```

Comment:
Do you need to remove obj dir as well?  I'm thinking you might be better off creating new test instances for each iteration so that you ensure that as people add new persisted artifacts, the tests remain effective.